ddp/ddp_pipeline/src/main/java/org/mskcc/cmo/ks/ddp/pipeline/util/DDPUtils.java

```diff
@@ -166,6 +174,43 @@ public static String resolvePatientCurrentAge(DDPCompositeRecord compositeRecord
         return anonymizePatientAge(age.intValue());
     }
 
+    public static String getAgeAtSeqDateLogging(String sampleId, String ageAtSeqDate, Long referenceInDays) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Sample '").append(sampleId).append("' resolves AGE_AT_SEQ_REPORTED_YEARS to null':  ")
+                .append("( AGE_AT_SEQ_REPORTED_YEARS=").append(ageAtSeqDate)
+                .append(", REFERENCE_DATE_DAYS=").append(referenceInDays)
+                .append(" )");
+        return builder.toString();
+    }
+
+    /**
+     * Resolve and anonymize patient age in years at date of sequencing. If either
+     * the patient date of birth or the date of sequencing is null this returns null.
+     *
+     * @param sampleId
+     * @param patientBirthDate String date of birth
+     * @return age in years at sequencing, null if we don't know patient date of birth or sequencing date
+     * @throws ParseException
+     */
+    public static String resolveAgeAtSeqDate(String sampleId, String patientBirthDate) throws ParseException {
+        Long birthDateInDays = getDateInDays(patientBirthDate);
+        Long referenceDateInDays = getDateInDays(sampleSeqDateMap.get(sampleId));
```

Comment:
Is this always the sample sequencing date? If so, I suggest an equally concrete name for this variable as "birthDateInDays" .. such as sampleSeqDateInDays